src/tools/clippy/tests/ui/format.stderr

```diff
@@ -39,12 +39,6 @@ error: useless use of `format!`
 LL |     let _ = format!("");
    |             ^^^^^^^^^^^ help: consider using `String::new()`: `String::new()`
 
-error: useless use of `format!`
-  --> $DIR/format.rs:29:5
-   |
-LL |     format!("{}", "foo");
-   |     ^^^^^^^^^^^^^^^^^^^^ help: consider using `.to_string()`: `"foo".to_string()`
-
```

Comment:
Hm, this breaks one of the main features of the `useless_format` lint in Clippy. Any chance to fix this in this PR?

Reply:
Hm, that expression will have been reduced to just `format!("foo")` before the late passes run, but that expression should trigger the same lint, right?

Reply:
Right, it should, but somehow it doesn't :thinking: @Alexendoo might be able to help here, because he worked on the format lints the most recently.

Reply:
Would it make sense to simply disable -Zflatten-format-args entirely when running clippy? I assume that would also be possible? Or is there a reason it must run with the exact same flags as regular rustc?

Reply:
Looks like that's possible. Did that in https://github.com/rust-lang/rust/pull/109999/commits/63d8758d35516d77dc53b5acaf0dc405f2297166

Reply:
If it's fine for that flag to stick around yeah that makes life easier for us, I can remove the other `rustc_ast::Expr` fall-backs in that case 👍